From a0894b52af4ee6cba42e93a2a756a2092ee46e48 Mon Sep 17 00:00:00 2001
From: Yauhen Kirylau <actionless.loveless@gmail.com>
Date: Fri, 11 Dec 2020 22:41:07 +0100
Subject: [PATCH] fix(x: main): do xrdb_load before setting cursor shape (fixes
 #127) (#129)

* fix(x: main): do xrdb_load before setting cursor shape (fixes #127)

* fix(x: main): do xrdb_load and xsetcursor after parsing cli args
---
 x.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git x.c x.c
index 4dcb168..5367d05 100644
--- x.c
+++ x.c
@@ -2111,7 +2111,6 @@ main(int argc, char *argv[])
 {
 	xw.l = xw.t = 0;
 	xw.isfixed = False;
-	xsetcursor(cursorshape);
 
 	ARGBEGIN {
 	case 'a':
@@ -2167,12 +2166,13 @@ main(int argc, char *argv[])
 	if (!opt_title)
 		opt_title = (opt_line || !opt_cmd) ? "xst" : opt_cmd[0];
 
+	xrdb_load();
+	xsetcursor(cursorshape);
 	setlocale(LC_CTYPE, "");
 	XSetLocaleModifiers("");
 	cols = MAX(cols, 1);
 	rows = MAX(rows, 1);
 	tnew(cols, rows);
-	xrdb_load();
 	xinit(cols, rows);
 	xsetenv();
 	selinit();
